### PR TITLE
What is subarray & subsequence?

### LINKED ISSUE #7036: What is Subarray and Subsequence ?
### Description

Description
- What is Subarray and subsequence?
- Difference between them.
- Program to print subarray and subsequence.

*Will add this File in C_Cpp/Data Structures/Array*

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by Nandani-koli**
<hr>

## Description 📜

What is Subarray and Subsequence.
Program to print subarray and subsequence
Fixes #7036 

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

### CODE REVIEW

File: C_CPP/Data Structures/Array/Subarray&Subsequence.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            cout<<endl;
+        }
+    }
+}
+```
+## Output :-
+![Subarray](https://user-images.githubusercontent.com/70843941/140601332-79d716b9-1910-4d1a-9117-1c1dd0388c73.png)
+
+# Subsequence
+*A subsequence is defined as a sequence that can be derived from an array by selecting zero or more elements without changing the order of the remaining elements.*
```
Comment (apeksha235) [inline C_CPP/Data Structures/Array/Subarray&Subsequence.md:48]: mention the formula for subsequence

File: C_CPP/Data Structures/Array/Subarray&Subsequence.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            }
+        }
+        cout<<endl;
+    }
+    return 0;
+}
+```
+## Output :-
+![Subsequence](https://user-images.githubusercontent.com/70843941/140635880-4d485767-49b8-45bb-aa1b-f4ec06438e32.png)
+
```
Comment (apeksha235) [inline C_CPP/Data Structures/Array/Subarray&Subsequence.md:97]: mention the time complexity as well

**Review by apeksha235 — CHANGES_REQUESTED** [review]
@Nandani-koli

**Comment by Nandani-koli** [comment]
@apeksha235 Thank you ma'am. I updated file.

**Review by deeqakkk — CHANGES_REQUESTED** [review]
Nandani, avoid using HTML stuff here(, it's not a good practice.
your code has so manyyyy nbsp, fix that.

**Comment by Nandani-koli** [comment]
@deeqakkk Thank you sir, I have made changes

**Comment by apeksha235** [comment]
@Nandani-koli amazing! 
@Saviour1001 please merge this

**Pull request merged**